target/linux/ath79/dts/qca9531_glinet_gl-xe300-nor-nand.dts

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca953x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "glinet,gl-xe300-nor-nand", "qca,qca9531";
+	model = "GL.iNet GL-XE300 Puli";
+
+	keys {
+		compatible = "gpio-keys";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&jtag_disable_pins>;
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 3 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power {
+			label = "green:power";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		lan {
+			gpios = <0x0f 0x0a 0x01>;
+			label = "gl-xe300:green:lan";
+		};
+
+		wan {
+			gpios = <0x0f 0x01 0x01>;
+			label = "gl-xe300:green:wan";
+		};
+
+		wlan {
+			gpios = <0x0f 0x0c 0x01>;
+			label = "gl-xe300:green:wlan";
+			linux,default-trigger = "phy0tpt";
+		};
+
+		lte {
+			gpios = <0x0f 0x0d 0x01>;
+			label = "gl-xe300:green:lte";
+		};
+	};
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&usb0 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "okay";
+
+	hub_port: port@1 {
+		reg = <1>;
+		#trigger-source-cells = <0>;
+	};
+};
+
+&usb_phy {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x40000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
+				reg = <0x40000 0x10000>;
+			};
+
+			art: partition@50000 {
+				label = "art";
+				reg = <0x50000 0x10000>;
+				read-only;
+			};
+
+			partition@60000 {
+				label = "kernel";
+				reg = <0x60000 0x400000>;
+			};
+
+			partition@460000 {
+				label = "nor_reserved";
+				reg = <0x460000 0xba0000>;
+			};
+		};
+	};
+
+	flash@1 {
+		compatible = "spi-nand";
+		reg = <1>;
+		spi-max-frequency = <25000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+            #address-cells = <1>;
+            #size-cells = <1>;
+
+			partition@0 {
+				label = "ubi";
+				reg = <0x0 0x8000000>;
+			};			
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&swphy4>;
+
+	mtd-mac-address = <&art 0x0>;
```

Comment:
Is it work in current master? possibly nvmem-cells should be used.

Reply:
Indeed, please convert to nvmem ...

Reply:
I'm not familiar with this format, do I need to write it like this?

&eth0 {
nvmem-cell = <&art 0x0>;
nvmem-cell-names = "mac-address";
}

This is what I understand from [https://elixir.bootlin.com/linux/v4.9/source/Documentation/devicetree/bindings/nvmem/nvmem.txt](nvmem documentation), it uses a 'provider'. In this DTS, the 'provider' is '&art' declared in '&spi-->flash@0-->art'.

Reply:
@adschm will @Vic063 proposed configuration work? I see commit 32adbfc uses the new approach, it looks like the proposed configuration is correct. I've copied it here as well:

```
&eth0 {
nvmem-cell = <&art 0x0>;
nvmem-cell-names = "mac-address";
}
```

Reply:
Ok, I'll fix that and this week-end I'll test it on the device to be sure it's good. I can't test right now, the device is not at home.